### DISCUSSION TITLE
Question: Why does IEnumerable<T>.ToOption resolves to Option<T>?

### DISCUSSION THREAD

**Discussion by LaszloLueck** [louthy/language-ext#1059] [General]
Hello,
why does `IEnumerable<T>.ToOption` transform to `Option<T>` and not to `Option<IEnumerable<T>>`?
And, if the `IEnumerable<T>` is not null, the Head (first entry) gives back as result?

I wrote myself a helper method that implement it more type-in, type-out.
    
```
[Pure]
public static Option<IEnumerable<T>> ToOption<T>(this IEnumerable<T>? self) => self.IsNull() ? Option<IEnumerable<T>>.None : self.ToSome();
```

From here, you can just resolve the option with IfNone....
e.g. 
```
IEnumerable<string>? nullList = null;
var res = nullList.ToOption().IfNone(Array.Empty<string>());
```
And res is 
a) an empty IEnumerable (if null) or the filled one.
b) the incoming IEnumerable as Value if it is not null

Thanks and Regards,
Laszlo

**Comment by louthy** [comment]
`ToOption()`, like all of the `To*` methods, is a [natural-transformation](https://en.wikipedia.org/wiki/Natural_transformation), not a lifting operation.  You can think of them as conversion functions.  Lifting is usually done by the constructor functions, like `Some(x)`, `Right(x)`, `FinSucc(x)` etc.  

In this case, the exact functionality you're looking for [is in `Optional(xs)`](https://github.com/louthy/language-ext/blob/main/LanguageExt.Core/Monads/Alternative%20Value%20Monads/Option/Option/Option.Prelude.cs#L118)

**Comment by LaszloLueck** [comment]
You´re absolutely right!
I made a small mistake, when i searching around for the right implementation.
I forgot to import
`using static LanguageExt.Prelude;`
So with that import works `Optional(nullable)` like a charm.

I think it is time to set
`global using static LanguageExt.Prelude;`
in the main.

Thanks a lot,

Laszlo
